Leftwm 0.5.2 fails to compile on 32 bit architecture

### Describe the bug

When compiling leftwm 0.5.2 for 32bit architecture, the compilation fails with the following error

```
Compiling xlib-display-server v0.1.3 (/builddir/leftwm-0.5.2/display-servers/xlib-display-server)
error[E0308]: mismatched types
   --> display-servers/xlib-display-server/src/xwrap/getters.rs:737:47
    |
737 |                 return Some(SliceIntoDockArea(slice).into());
    |                             ----------------- ^^^^^ expected `&[i64]`, found `&[i32]`
    |                             |
    |                             arguments to this struct are incorrect
    |
    = note: expected reference `&[i64]`
               found reference `&[i32]`
note: tuple struct defined here
   --> display-servers/xlib-display-server/src/xwrap/getters.rs:821:8
    |
821 | struct SliceIntoDockArea<'a>(&'a [i64]);
    |        ^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> display-servers/xlib-display-server/src/xwrap/getters.rs:753:47
    |
753 |                 return Some(SliceIntoDockArea(slice).into());
    |                             ----------------- ^^^^^ expected `&[i64]`, found `&[i32]`
    |                             |
    |                             arguments to this struct are incorrect
    |
    = note: expected reference `&[i64]`
               found reference `&[i32]`
note: tuple struct defined here
   --> display-servers/xlib-display-server/src/xwrap/getters.rs:821:8
    |
821 | struct SliceIntoDockArea<'a>(&'a [i64]);
    |        ^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `xlib-display-server` (lib) due to 2 previous errors
```


### Expected behavior (optional)

The code should compile on 32 bit architecture

### Steps to reproduce (optional)

Clone the repo from tag 0.5.2 and try to compile on 32bit

### Relevant log output (optional)

```text
Compiling xlib-display-server v0.1.3 (/builddir/leftwm-0.5.2/display-servers/xlib-display-server)
error[E0308]: mismatched types
   --> display-servers/xlib-display-server/src/xwrap/getters.rs:737:47
    |
737 |                 return Some(SliceIntoDockArea(slice).into());
    |                             ----------------- ^^^^^ expected `&[i64]`, found `&[i32]`
    |                             |
    |                             arguments to this struct are incorrect
    |
    = note: expected reference `&[i64]`
               found reference `&[i32]`
note: tuple struct defined here
   --> display-servers/xlib-display-server/src/xwrap/getters.rs:821:8
    |
821 | struct SliceIntoDockArea<'a>(&'a [i64]);
    |        ^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> display-servers/xlib-display-server/src/xwrap/getters.rs:753:47
    |
753 |                 return Some(SliceIntoDockArea(slice).into());
    |                             ----------------- ^^^^^ expected `&[i64]`, found `&[i32]`
    |                             |
    |                             arguments to this struct are incorrect
    |
    = note: expected reference `&[i64]`
               found reference `&[i32]`
note: tuple struct defined here
   --> display-servers/xlib-display-server/src/xwrap/getters.rs:821:8
    |
821 | struct SliceIntoDockArea<'a>(&'a [i64]);
    |        ^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0308`.
error: could not compile `xlib-display-server` (lib) due to 2 previous errors
```


### OS / Distro

Linux 6.11.3_2 #1 SMP PREEMPT_DYNAMIC Tue Oct 15 02:54:10 UTC 2024 x86_64 GNU/Linux

### Additional System Information (optional)

_No response_

### LeftWM Check

```text
N/A
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.5.3]
+
+### Fixed
+
+- Fixed compilation issue for xlib-display-server for 32-bit systems
+
 ## [0.5.2]
 
 ## CVEs

--- a/display-servers/x11rb-display-server/Cargo.toml
+++ b/display-servers/x11rb-display-server/Cargo.toml
@@ -1,14 +1,14 @@
 [package]
 name = "x11rb-display-server"
 description = "x11 backend for leftwm using pure rust + xcb specification"
-version = "0.1.1"
+version = "0.1.2"
 license = "MIT"
 edition = "2021"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-leftwm-core = { path = "../../leftwm-core", version = '0.5.1' }
+leftwm-core = { path = "../../leftwm-core", version = '0.5.2' }
 futures = "0.3.21"
 tracing = "0.1.36"
 tokio = { version = "1.2.0", features = [ "sync", "time" ] }

--- a/display-servers/xlib-display-server/Cargo.toml
+++ b/display-servers/xlib-display-server/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "xlib-display-server"
-version = "0.1.3"
+version = "0.1.4"
 description = "A display server library for LeftWM" 
 license = "MIT"
 edition = "2021"
@@ -8,7 +8,7 @@ edition = "2021"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-leftwm-core = { path = "../../leftwm-core", version = '0.5.1' }
+leftwm-core = { path = "../../leftwm-core", version = '0.5.2' }
 x11-dl = "2.18.4"
 futures = "0.3.21"
 tracing = "0.1.36"

--- a/display-servers/xlib-display-server/src/xwrap/getters.rs
+++ b/display-servers/xlib-display-server/src/xwrap/getters.rs
@@ -818,7 +818,11 @@ impl From<XWindowAttributesIntoScreen<'_>> for Screen<XlibWindowHandle> {
     }
 }
 
+#[cfg(target_pointer_width = "64")]
 struct SliceIntoDockArea<'a>(&'a [i64]);
+
+#[cfg(target_pointer_width = "32")]
+struct SliceIntoDockArea<'a>(&'a [i32]);
 
 impl From<SliceIntoDockArea<'_>> for DockArea {
     fn from(val: SliceIntoDockArea<'_>) -> Self {

--- a/leftwm-core/Cargo.toml
+++ b/leftwm-core/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "leftwm-core"
-version = "0.5.1"
+version = "0.5.2"
 rust-version = "1.74.0"
 authors = ["Lex Childs <lexchilds@gmail.com>"]
 categories = ["data-structures", "gui"]

--- a/leftwm-macros/Cargo.toml
+++ b/leftwm-macros/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "leftwm-macros"
-version = "0.5.1"
+version = "0.5.2"
 rust-version = "1.74.0"
 authors = ["Lex Childs <lexchilds@gmail.com>"]
 categories = ["development-tools::procedural-macro-helpers"]

--- a/leftwm-watchdog/Cargo.toml
+++ b/leftwm-watchdog/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "leftwm-watchdog"
-version = "0.5.2"
+version = "0.5.3"
 rust-version = "1.74.0"
 authors = ["Lex Childs <lexchilds@gmail.com>"]
 categories = ["gui"]

--- a/leftwm/Cargo.toml
+++ b/leftwm/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "leftwm"
-version = "0.5.2"
+version = "0.5.3"
 rust-version = "1.74.0"
 authors = ["Lex Childs <lexchilds@gmail.com>"]
 categories = ["gui"]
@@ -23,8 +23,8 @@ once_cell = "1.13.0"
 futures = "0.3.21"
 git-version = "0.3.5"
 lefthk-core = { version = '0.2', optional = true }
-leftwm-core = { path = "../leftwm-core", version = '0.5.1' }
-leftwm-macros = {path = "../leftwm-macros", version = '0.5.1'}
+leftwm-core = { path = "../leftwm-core", version = '0.5.2' }
+leftwm-macros = {path = "../leftwm-macros", version = '0.5.2'}
 leftwm-layouts = "0.9.1"
 liquid = "0.26.0"
 mio = "1.0.2"

